In [407]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [3]:
gps = pd.read_csv('googleplaystore.csv')
gps


App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[10841 rows x 13 columns]

# 1.DISPLAY TOP 5 ROWS OF THE DATASET

In [408]:
#gps.head()
#gps[:5]
#gps.loc[:5]
gps.iloc[:5]


App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

# 2.DISPLAY LAST 3 ROWS OF THE DATASET

In [13]:
#gps.tail(3)          # method 1
lr = len(gps)-3
#gps[lr:]             # method 2
#gps.loc[lr:]         # method 3
gps.iloc[lr:]         # method 4

App             Category  \
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

# 3.  FIND SHAPE NO. OF ROWS AND NO. OF COLUMNS

In [23]:
print('NO. OF ROWS:',(gps.shape[0]))
print('NO. OF COLUMNS:',(gps.shape[1]))


NO. OF ROWS: 10841
NO. OF COLUMNS: 13


# 4. GET INFORMATION ABOUT OUR DATASET

In [24]:
gps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


# 5.GET OVERALL STATISTICS ABOUT THE DATAFRAME

In [410]:
gps.describe(include = 'all')


App Category       Rating Reviews                Size    Installs  \
count    10358    10358  8893.000000   10358               10358       10358   
unique    9660       34          NaN    6002                 462          22   
top     ROBLOX   FAMILY          NaN       0  Varies with device  1,000,000+   
freq         9     1943          NaN     593                1526        1488   
mean       NaN      NaN     4.189542     NaN                 NaN         NaN   
std        NaN      NaN     0.545452     NaN                 NaN         NaN   
min        NaN      NaN     1.000000     NaN                 NaN         NaN   
25%        NaN      NaN     4.000000     NaN                 NaN         NaN   
50%        NaN      NaN     4.300000     NaN                 NaN         NaN   
75%        NaN      NaN     4.500000     NaN                 NaN         NaN   
max        NaN      NaN    19.000000     NaN                 NaN         NaN   

         Type  Price Content Rating Genres    Last Updated  \
count   10357  10358          10357  10358           10358   
unique      3     93              6    120            1378   
top      Free      0       Everyone  Tools  August 3, 2018   
freq     9591   9592           8382    842             298   
mean      NaN    NaN            NaN    NaN             NaN   
std       NaN    NaN            NaN    NaN             NaN   
min       NaN    NaN            NaN    NaN             NaN   
25%       NaN    NaN            NaN    NaN             NaN   
50%       NaN    NaN            NaN    NaN             NaN   
75%       NaN    NaN            NaN    NaN             NaN   
max       NaN    NaN            NaN    NaN             NaN   

               Current Ver Android Ver  
count                10350       10355  
unique                2832          33  
top     Varies with device  4.1 and up  
freq                  1302        2379  
mean                   NaN         NaN  
std                    NaN         NaN  
min                    NaN         NaN  
25%                    NaN         NaN  
50%                    NaN         NaN  
75%                    NaN         NaN  
max                    NaN         NaN

# 6. total number of app titles contain Astrology

In [54]:
As = []
for i in gps.App:
    if 'Astrology' in i:
        As.append(i)
print('App name which contains Astrology:',As)
print('Total number of app title contain Astrology:',len(As))


App name which contains Astrology: ['Horoscopes – Daily Zodiac Horoscope and Astrology', 'သိင်္ Astrology - Min Thein Kha BayDin', 'iHoroscope - 2018 Daily Horoscope & Astrology']
Total number of app title contain Astrology: 3


# 7. find avg app rating

In [60]:
print('overall app rating is:',gps.Rating.mean())


overall app rating is: 4.193338315362448


# 8.Total number of unique category:

In [66]:
pd.DataFrame(gps.Category.unique())


0
0        ART_AND_DESIGN
1     AUTO_AND_VEHICLES
2                BEAUTY
3   BOOKS_AND_REFERENCE
4              BUSINESS
5                COMICS
6         COMMUNICATION
7                DATING
8             EDUCATION
9         ENTERTAINMENT
10               EVENTS
11              FINANCE
12       FOOD_AND_DRINK
13   HEALTH_AND_FITNESS
14       HOUSE_AND_HOME
15   LIBRARIES_AND_DEMO
16            LIFESTYLE
17                 GAME
18               FAMILY
19              MEDICAL
20               SOCIAL
21             SHOPPING
22          PHOTOGRAPHY
23               SPORTS
24     TRAVEL_AND_LOCAL
25                TOOLS
26      PERSONALIZATION
27         PRODUCTIVITY
28            PARENTING
29              WEATHER
30        VIDEO_PLAYERS
31   NEWS_AND_MAGAZINES
32  MAPS_AND_NAVIGATION
33                  1.9

In [68]:
print('Total number of unique categorie:',len(gps.Category.unique()))


Total number of unique categorie: 34


# 9.Which category getting the highest avg rating

In [166]:
#(gps.Category.unique())
hc = gps.groupby('Category').Rating.mean().nlargest()
hc


Category
1.9                    19.000000
EVENTS                  4.435556
EDUCATION               4.389032
ART_AND_DESIGN          4.358065
BOOKS_AND_REFERENCE     4.346067
Name: Rating, dtype: float64

### here we can see 1.9 Category has an avg rating is 19.00, but as we know rating should not be more than 5.00
so, in my opinion 'EVENT' category has a highest avg rating of 4.435556.

In [172]:
print('The Category which has highest avg rating :',hc.index[1])


The Category which has highest avg rating : EVENTS


# 10. find total no. of app having 5 star rating

In [211]:
print('Total no. of app having 5 star rating is:', len(gps.query('Rating==5.0')))


Total no. of app having 5 star rating is: 274


# 11. find avg value of reviews

In [210]:
for i in gps.Reviews:
    if 'M' in i:
        print(i)
        

3.0M


In [213]:
gps.Reviews.replace('3.0M',3000000,inplace=True)


In [222]:
avg = gps.Reviews.astype('float').describe()['mean']
avg


444388.6535374965

In [224]:
print('avg value of reviews:',avg)


avg value of reviews: 444388.6535374965


# 12. find total number of free and paid apps

In [238]:
print('total free apps in dataset :',gps.query('Type=="Free"').shape[0])
print('total paid apps in dataset :',gps.query('Type=="Paid"').shape[0])


total free apps in dataset : 10039
total paid apps in dataset : 800


# 13. which app has maximum reviews

In [335]:
fr = gps.Reviews.astype('float').nlargest(15)
fr


2544    78158306.0
3943    78128208.0
336     69119316.0
3904    69109672.0
2604    66577446.0
2545    66577313.0
3909    66509917.0
382     56646578.0
335     56642847.0
1879    44893888.0
1670    44891723.0
3986    44881447.0
4005    42916526.0
1872    27725352.0
1750    27724094.0
Name: Reviews, dtype: float64

In [336]:
gps.loc[[fr.index[0]]].App.values[0]


'Facebook'

In [337]:
print('App has maximum review:',gps.loc[[fr.index[0]]].App.values[0])


App has maximum review: Facebook


# 14. display top 5 app having highest review

In [338]:
fr

2544    78158306.0
3943    78128208.0
336     69119316.0
3904    69109672.0
2604    66577446.0
2545    66577313.0
3909    66509917.0
382     56646578.0
335     56642847.0
1879    44893888.0
1670    44891723.0
3986    44881447.0
4005    42916526.0
1872    27725352.0
1750    27724094.0
Name: Reviews, dtype: float64

In [349]:
for i in range(5):
    print(gps.loc[fr.index[i]].App)
    

Facebook
Facebook
WhatsApp Messenger
WhatsApp Messenger
Instagram


### but here we get repeated apps, we have to do something to get top 5 app having highest reviwes with different name

In [347]:
had = []
for i in range(15):
    ha = gps.loc[fr.index[i]].App
    had.append(ha)
pd.DataFrame(had).drop_duplicates().head()


0
0                                  Facebook
2                        WhatsApp Messenger
4                                 Instagram
7  Messenger – Text and Video Chat for Free
9                            Clash of Clans

# 15. find avg rating of free and paid apps

In [355]:
print('Avg rating of free apps:',gps.query('Type=="Free"').Rating.mean())
print('Avg rating of paid apps:',gps.query('Type=="Paid"').Rating.mean())


Avg rating of free apps: 4.1824254136973105
Avg rating of paid apps: 4.261500815660683


# 16. Display top 5 apps having maximum installs

In [388]:
gps.loc[[10472]]


App Category  Rating  Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0  3000000   

         Size  Installs Type     Price Content Rating             Genres  \
10472  1,000+  4000000+    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [387]:
gps.Installs.replace('1000000+','4000000+',inplace = True)

In [406]:
mi = gps.Installs.sort_values(ascending=False).head()
mi5 = []
for i in range(5):
    mi5.append(gps.loc[[mi.index[i]]].App.values)
pd.DataFrame(mi5)


0
0                       Candy Crush Saga
1                       Google Translate
2                         Microsoft Word
3           Flipboard: News For Our Time
4  Google Duo - High Quality Video Calls